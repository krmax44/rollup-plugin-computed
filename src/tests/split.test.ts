import type { Computers } from '..';
import { roll } from './utils';

describe('split', () => {
	it('splits by default', async () => {
		expect.assertions(2);

		const computers: Computers = {
			test() {
				return { it: 'works' };
			}
		};

		const output = await roll('basic.js', computers);
		const file: any = output.find(c => c.fileName === 'test.js');

		expect(output.length).toBe(2);
		expect(file.code).toMatchSnapshot();
	});

	it("doesn't split when asked to", async () => {
		expect.assertions(2);

		const computers: Computers = {
			test: {
				split: false,
				fn() {
					return { it: 'works' };
				}
			}
		};

		const output = await roll('basic.js', computers);
		const file: any = output.find(c => c.fileName === 'basic.js');

		expect(output.length).toBe(1);
		expect(file.code).toMatchSnapshot();
	});
});
