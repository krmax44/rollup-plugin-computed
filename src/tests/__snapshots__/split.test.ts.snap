// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`split doesn't split by default 1`] = `
"var test = {\\"it\\":\\"works\\"};

console.log(test);
"
`;

exports[`split splits when asked to 1`] = `
"var test = {\\"it\\":\\"works\\"};

export default test;
"
`;
